server/broadcast.go

```diff
@@ -347,13 +359,30 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 	// send segment to the orchestrator
 	glog.V(common.DEBUG).Infof("Submitting segment nonce=%d manifestID=%s seqNo=%d orch=%s", nonce, cxn.mid, seg.SeqNo, sess.OrchestratorInfo.Transcoder)
 
-	res, err := SubmitSegment(sess, seg, nonce)
-	if err != nil || res == nil {
+	res, tErr := SubmitSegment(sess, seg, nonce)
+	// If TicketParams have expired, refresh them
+	if tErr != nil && tErr == pm.ErrTicketParamsExpired {
```

Comment:
Could some of this extra nesting + error/nil checking be avoided if we use `sender.ValidateTicketParams()` here instead? Something like:

```
if sess.Sender != nil {
	err := sess.Sender.ValidateTicketParams(params)
	if err != nil {
		if err != pm.ErrTicketParamsExpired {
			return nil, err
		}

		sess, err = refreshSession(sess)
		if err != nil {
			cxn.sessManager.removeSession(sess)
			return nil, err
		}
	}
}

// Call SubmitSegment
```

Reply:
fixed in c274f67c5296ea35f0c635d0b86ef5780506cf4a & ebb7532edd782df1b2352b1beb5e46159687474b